### PR TITLE
resequence

### PR DESCRIPTION
**Pull request by jpp-odoo**

### CODE REVIEW

File: addons/web/static/src/views/relational_model/dynamic_group_list.js [hunk 1]
```diff
@@ -100,6 +101,7 @@ export class DynamicGroupList extends DynamicList {
         const [id] = await this.model.orm.create(this.groupByField.relation, [groupData], {
             context: this.context,
         });
+        const [lastGroup] = this.groups.slice(-1);
```
Comment (mcm-odoo) [inline addons/web/static/src/views/relational_model/dynamic_group_list.js]:
```suggestion
        const lastGroup = this.groups.at(-1);
```

File: addons/web/static/src/views/relational_model/dynamic_group_list.js [hunk 2]
```diff
@@ -165,6 +175,21 @@ export class DynamicGroupList extends DynamicList {
         return Promise.all(proms);
     }
 
+    async resequence(movedGroupId, targetGroupId) {
+        const resModel = isRelational(this.groupByField)
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/dynamic_group_list.js]: only for many2one

File: addons/web/static/src/views/relational_model/dynamic_group_list.js [hunk 3]
```diff
@@ -139,7 +141,15 @@ export class DynamicGroupList extends DynamicList {
             displayName: groupName,
         };
 
-        this.groups.push(this._createGroupDatapoint(data));
+        const group = this._createGroupDatapoint(data);
+        this.groups.push(group);
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/dynamic_group_list.js]:
```suggestion
        if (lastGroup) {
            const groups = [...this.groups, group];
            ...
            this.groups = await ...;
        } else {
            this.groups.push(group);
        }
```

File: addons/web/static/src/views/relational_model/dynamic_group_list.js [hunk 4]
```diff
@@ -139,7 +141,15 @@ export class DynamicGroupList extends DynamicList {
...
+
+        if (lastGroup) {
+            const resModel = isRelational(this.groupByField)
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/dynamic_group_list.js]: only many2one

Comment (aab-odoo) [reply addons/web/static/src/views/relational_model/dynamic_group_list.js]: throw error if called and not grouped by many2one. Do it at both places, at the model entry point

File: addons/web/static/src/views/relational_model/dynamic_list.js [hunk 5]
```diff
@@ -275,4 +276,108 @@ export class DynamicList extends DataPoint {
...
+
+        // Find indices
+        const fromIndex = dataPoints.findIndex((r) => r.id === movedId);
+        let toIndex = 0;
+        if (targetId !== null) {
+            const targetIndex = dataPoints.findIndex((r) => r.id === targetId);
+            toIndex = fromIndex > targetIndex ? targetIndex + 1 : targetIndex;
+        }
+
+        const getSequence = (rec) => {
```
Comment (jpp-odoo) [inline addons/web/static/src/views/relational_model/dynamic_list.js]:
```suggestion
        const getSequence = (dp) => {
```

File: addons/web/static/src/views/relational_model/dynamic_list.js [hunk 6]
```diff
@@ -275,4 +276,108 @@ export class DynamicList extends DataPoint {
...
+        const result = await this.model.orm.read(resModel, resIds, [handleField], kwargs);
+        for (const recordData of result) {
+            const dp = dataPoints.find(
+                (r) => r.resId === recordData.id || r.value === recordData.id
+            );
+            const value = { [handleField]: recordData[handleField] };
+            if (dp instanceof Record) {
+                dp._applyValues(recordData);
+            } else {
+                Object.assign(dp, value);
```
Comment (jpp-odoo) [inline addons/web/static/src/views/relational_model/dynamic_list.js]:
```suggestion
                dp[handleField] = recordData[handleField]
```

**Pull request merged**